OpenKeychain/src/main/AndroidManifest.xml

```diff
@@ -76,6 +76,8 @@
 
     <!-- other group (for free) -->
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
```

Comment:
`NetworkInfo` will require `ACCESS_NETWORK_STATE` permission ([here](http://stackoverflow.com/a/2326816)).

You're right about `ACCESS_WIFI_STATE` being redundant. Just to be sure I will run a test after removing it.